influxdb/line_protocol.py

```diff
@@ -23,19 +23,36 @@ def _convert_timestamp(timestamp, precision=None):
     if isinstance(timestamp, datetime):
         if not timestamp.tzinfo:
             timestamp = UTC.localize(timestamp)
-        ns = (timestamp - EPOCH).total_seconds() * 1e9
-        if precision is None or precision == 'n':
-            return ns
-        elif precision == 'u':
-            return ns / 1e3
-        elif precision == 'ms':
-            return ns / 1e6
-        elif precision == 's':
-            return ns / 1e9
-        elif precision == 'm':
-            return ns / 1e9 / 60
-        elif precision == 'h':
-            return ns / 1e9 / 3600
+        try:
+            ns = (timestamp - EPOCH).total_seconds() * 1e9
+            if precision is None or precision == 'n':
+                return ns
+            elif precision == 'u':
+                return ns / 1e3
+            elif precision == 'ms':
+                return ns / 1e6
+            elif precision == 's':
+                return ns / 1e9
+            elif precision == 'm':
+                return ns / 1e9 / 60
+            elif precision == 'h':
+                return ns / 1e9 / 3600
+        except AttributeError:
+            td = (timestamp - EPOCH)
+            ns = ((td.microseconds + (td.seconds + td.days * 24 * 3600) *
+                   10 ** 6) / 10 ** 6) * 1e9
+            if precision is None or precision == 'n':
```

Comment:
this section can be removed from the try/except, as it is common to both cases once you've defined `ns`